### ISSUE TITLE
False positive PT009 for unittest.TestCase.assertRaises

### ISSUE THREAD

**Issue by jamesbraza** [m-burst/flake8-pytest-style#214]
# Bug report

## What's wrong

[PT009](https://github.com/m-burst/flake8-pytest-style/blob/master/docs/rules/PT009.md) is about moving away from `unittest.TestCase`'s assertion methods.  I think for many methods (e.g. `assertEqual`, `assertNotEqual`) this jump makes sense.  However, this jump doesn't make sense for `assertRaises`:

```python
from unittest import TestCase


class TestFoo(TestCase):
    def test_method(self) -> None:
        with self.assertRaises(ValueError):
            raise ValueError("BOOM")
```

`flake8-pytest-style` will throw:

```
PT009 use a regular assert instead of unittest-style 'assertRaises'
```

This is incorrect, as a regular `assert` can't be applied here.  Therefore I think PT009 should not be raised.

## How it should work

`flake8-pytest-style` could do one or both of:
- Not raise a PT009 for `assertRaises`
- Suggest to switch to `pytest.raises`
    - Maybe also report [PT011](https://github.com/m-burst/flake8-pytest-style/blob/master/docs/rules/PT011.md)

## System information

* Operating system: macOS Monterey version 12.6
* Python version: 3.10.9
* flake8 version: 6.0.0
* flake8-pytest-style version: 1.6.0

**Comment by m-burst** [comment]
Hi James,

Thanks for reporting this.
You are correct, it would be better to suggest `pytest.raises` as a replacement for `assertRaises` and not a regular assert.

**Comment by m-burst** [comment]
I've just released version 1.7.0 with this change.

**Comment by jamesbraza** [comment]
Thanks @m-burst !  One comment on the PR in [this line](https://github.com/m-burst/flake8-pytest-style/pull/217/files#diff-cd35a298a38735bb9b011df92789f058f3a0c6b382701f16e7805c1e03cc2311R48).

```python
    'assertRaisesRegexp'
```

This seems to be a deprecated alias, the new alias is `assertRaisesRegex`.

Any thoughts on adding that to the code?

**Comment by m-burst** [comment]
I've opened a separate issue to make sure that none of the unittest-style assertions are missing in our checks.

**Comment by m-burst** [comment]
Hi @jamesbraza,
This is now fixed in v1.7.1
